### PR TITLE
Add more details(PSK or 802.1X) in network_stats() crypto

### PR DESCRIPTION
**Pull request by qingxp9**
It will be more helpful to show the auth: PSK or 802.1X. 

And because they may support one of the WPA or WPA2, or both,  I just added in the crypto array. 

Outputs like these:
```
t = sniff(iface="wlan0",  filter="subtype probe-resp or subtype beacon", timeout=0.5)
for i in range(len(t)):
  p = t[i]
  if p.subtype == 8: #Beacon
    print( p[Dot11Beacon].network_stats()["crypto"])
  if p.subtype == 5: #ProbeResp
    print( p[Dot11ProbeResp].network_stats()["crypto"])

{'WPA2/PSK'}
{'WPA2/PSK'}
{'WEP'}
{'OPN'}
{'WPA2/802.1X'}
{'OPN'}
{'WPA2/802.1X'}
{'OPN'}
{'WPA2/PSK'}
{'WEP'}
{'WPA/PSK', 'WPA2/PSK'}
{'WPA2/802.1X'}
{'WPA2/PSK'}
{'WPA2/802.1X'}
{'WPA2/PSK'}
{'WPA2/PSK'}
{'WPA/PSK', 'WPA2/PSK'}
{'WPA/PSK', 'WPA2/PSK'}
{'WPA2/PSK'}
{'WPA2/802.1X'}

```

### CODE REVIEW

File: scapy/layers/dot11.py [hunk 1]
```diff
@@ -640,11 +640,24 @@ def network_stats(self):
...
+
+                crypto.add("WPA2/%s" %(auth))
             elif p.ID == 221:
                 if isinstance(p, Dot11EltMicrosoftWPA) or \
                         p.info.startswith(b'\x00P\xf2\x01\x01\x00'):
-                    crypto.add("WPA")
+                    akmsuite_types = {
+                        0x01: "802.1X",
+                        0x02: "PSK"
+                    }
```
Comment (gpotter2) [inline scapy/layers/dot11.py]: Could you move the dict up in the function, so that it’s only defined once ?

Comment (gpotter2) [reply scapy/layers/dot11.py]: Also, it’s technically possible to have length = 0 (no akm_suites) available. You should check it’s not empty first

Comment (qingxp9) [reply scapy/layers/dot11.py]: Thanks for the suggestion, I have done.

**Comment by qingxp9** [comment]
It's necessary to show an AP is authorized by 802.1X or PSK, I just gave an example. Or you'd like other way to do this.

File: scapy/layers/dot11.py [hunk 2]
```diff
@@ -640,11 +645,19 @@ def network_stats(self):
             elif isinstance(p, Dot11EltRates):
                 summary["rates"] = p.rates
             elif isinstance(p, Dot11EltRSN):
-                crypto.add("WPA2")
+                if hasattr(p, "akm_suites"):
+                    auth = akmsuite_types.get(p.akm_suites[0].suite)
+                    crypto.add("WPA2/%s" % (auth))
```
Comment (gpotter2) [inline scapy/layers/dot11.py]: The parenthesis are redundant

File: scapy/layers/dot11.py [hunk 3]
```diff
@@ -640,11 +645,19 @@ def network_stats(self):
             elif isinstance(p, Dot11EltRates):
                 summary["rates"] = p.rates
             elif isinstance(p, Dot11EltRSN):
-                crypto.add("WPA2")
+                if hasattr(p, "akm_suites"):
```
Comment (gpotter2) [inline scapy/layers/dot11.py]: This is always true. I meant to rather use:
`if p.akm_suites:`

Comment (qingxp9) [reply scapy/layers/dot11.py]: done

File: scapy/layers/dot11.py (hunk 2 above)
Comment (gpotter2) [reply scapy/layers/dot11.py]: This is still not fixed :/

Comment (qingxp9) [reply scapy/layers/dot11.py]: I have replaced the two places with `if p.akm_suites:`, have I misunderstand something?
I have no idea why one check falls, I open another #1917, but still meet fails.

**Pull request closed without merging**